app/components/course-page/completed-step-notice.ts

```diff
@@ -42,6 +42,22 @@ export default class CompletedStepNoticeComponent extends Component<Signature> {
     return this.coursePageState.nextStep;
   }
 
+  get nextStepModels() {
+    if (this.nextStep?.type === 'BaseStagesCompletedStep') {
+      return this.nextStep.routeParams.models;
+    }
+
+    return this.activeStep?.routeParams.models;
+  }
+
+  get nextStepRoute() {
+    if (this.nextStep?.type === 'BaseStagesCompletedStep') {
+      return this.nextStep.routeParams.route;
+    }
+
+    return this.activeStep?.routeParams.route;
```

Comment:
@libmartinito the naming here is confusing - we already have a "NextStep" concept, so having `nextStep` route return `activeStep` is inconsistent.

Reply:
Still thinking of what a good name/structure could be here...

Reply:
Okay, best I can come up with:

```typescript
get stepForNextOrActiveStepButton() {
  return this.nextStep?.type === 'BaseStagesCompletedStep' ? this.nextStep : this.activeStep;
}
```

And let's rename the button to say `data-test-next-or-active-step-button`.